github/repos_contents.go

```diff
@@ -119,6 +120,10 @@ func (s *RepositoriesService) DownloadContents(owner, repo, filepath string, opt
 	return nil, fmt.Errorf("No file named %s found in %s", filename, dir)
 }
 
+func escapeURLPath(path string) string {
+	return template.URLQueryEscaper(path)
```

Comment:
so this is nice in that it works with go1.4.  Internally, `html/template.URLQueryEscaper` just calls `text/template.URLQueryEscaper` which in turn calls `net/url.QueryEscape`, so let's just use that directly here.

I'm also a little bit concerned about *over* escaping... url.QueryEscape escapes [all of these characters](https://github.com/golang/go/blob/aa22c42d704f9c6d9d8e65f972ac819678629ff2/src/net/url/url.go#L123) which is probably more than we need.  At least escaping "/" is not harmful (to wit: https://api.github.com/repos/MattBlack85/Golearning/contents/golang_tour%2Fexcercise.tour%2325.go).  How would you feel about trying to create a test file that uses all of the characters listed above (or at least as many as your OS will allow) to verify that GitHub unescapes them all properly (I feel fairly certain it does, but it'd be nice to confirm).

It's also interesting to note that the GitHub API itself does not escape the '#' in URLs it outputs (e.g: the response from https://api.github.com/repositories/12052373/contents/golang_tour). /cc @pengwynn @izuzak